api/filters.go

```diff
@@ -52,6 +58,19 @@ func (api *API) createFilter(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	if v.State != published && !dprequest.IsCallerPresent(ctx) {
+		api.respond.Error(
+			ctx,
+			w,
+			http.StatusNotFound,
+			Error{
+				err:     errors.New("unauthenticated request on unpublished dataset"),
+				message: "dataset not found",
```

Comment:
The message seems a little misleading. Perhaps something like this would prevent confusion:
```suggestion
				message: "dataset only available to authorised users",
```

Reply:
That was on purpose, I was thinking the public shouldn't be aware of unplublished datasets? The full error message with details regarding authentication does get logged for our purposes though (`err` gets logged, `message` is returned to caller)

Reply:
Yes I realise that. I just think it would be better not to give the caller a false impression. Fair enough that they shouldn't have access to the dataset. But being aware that it exists but is not published should be ok. So perhaps something like "dataset not published" might be better?